### PR TITLE
Changed the implementation of label extraction from the stack traces

### PR DESCRIPTION
**Pull request by Andarist**

### CODE REVIEW

File: packages/react/src/emotion-element.js [hunk 1]
```diff
@@ -42,16 +42,23 @@ export const createEmotionProps = (type: React.ElementType, props: Object) => {
   if (process.env.NODE_ENV !== 'production') {
     const error = new Error()
     if (error.stack) {
-      // chrome
+      // hooks|classes|ssr
       let match = error.stack.match(
-        /at (?:Object\.|Module\.|)(?:jsx|createEmotionProps).*\n\s+at (?:Object\.|)([A-Z][A-Za-z0-9$]+) /
+        /(^.+)\n.*(renderWithHooks|finishClassComponent|processChild)/m
```
Comment (mitchellhamilton) [inline packages/react/src/emotion-element.js:47]: It seems like only V8 is the only engine to include the class name in the stack trace, the others only include `render` so using finishClassComponent is probably not gonna go well and it looks like processChild is used for both class and function components so maybe we should not match `ComponentName.render` for the V8 case?

Comment (Andarist) [reply packages/react/src/emotion-element.js:47]: So just to be sure - are you propose matching lines with `ComponentName` (but without `.render`!) followed by lines with `renderWithHooks|processChild`?

Comment (mitchellhamilton) [reply packages/react/src/emotion-element.js:47]: Yes, unless there's another way we could not match class components?

**Comment by Andarist** [comment]
This got superseded by https://github.com/emotion-js/emotion/pull/2534

**Pull request closed without merging**